### PR TITLE
`infernal`: add version `1.1.5`

### PR DESCRIPTION
**Pull request by luke-dt**
Small fixes to dependencies for py-bakta to build with apple-clang on apple silicon machines

### CODE REVIEW

File: var/spack/repos/builtin/packages/blast-plus/package.py [hunk 1]
```diff
@@ -166,4 +168,15 @@ def configure_args(self):
         else:
             config_args.append("--without-python")
 
+        if spec.satisfies("-openmp"):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/blast-plus/package.py]:
```suggestion
        if spec.satisfies("~openmp"):
```

File: var/spack/repos/builtin/packages/blast-plus/package.py [hunk 2]
```diff
@@ -79,6 +80,7 @@ def patch(self):
     depends_on("bzip2", when="+bzip2")
     depends_on("lzo", when="+lzo")
     depends_on("pcre", when="+pcre")
+    depends_on("llvm-openmp", when="%apple-clang")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/blast-plus/package.py]: Should this include the new `openmp` variant?

```suggestion
    depends_on("llvm-openmp", when="+openmp%apple-clang")
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the new version sha256.  See comments for some changes.

**Comment by tldahlgren** [comment]
@weijianwen @RMeli Do you want to comment on the changes to your packages?

File: var/spack/repos/builtin/packages/py-biopython/package.py [hunk 3]
```diff
@@ -28,6 +28,7 @@ class PyBiopython(PythonPackage):
     version("1.65", sha256="6d591523ba4d07a505978f6e1d7fac57e335d6d62fb5b0bcb8c40bdde5c8998e")
 
     depends_on("c", type="build")  # generated
+    depends_on("libxcrypt", when="%apple-clang")
```
Comment (RMeli) [inline var/spack/repos/builtin/packages/py-biopython/package.py]: I don't see this requirement in the setup.py and I previously built BioPython successfully on macOS without this library. Can you please clarify why this change is necessary?

Comment (luke-dt) [reply var/spack/repos/builtin/packages/py-biopython/package.py]: so I'm also a little confused, because it was having issues finding a header that it appears that biopython code doesn't ask for it. However, I was having the following error when trying to build biopython on macOS:

<details><summary>Error message</summary>

<pre>
     876      building 'Bio.Align._aligners' extension
     877      creating build/temp.macosx-14.0-arm64-cpython-38
     878      creating build/temp.macosx-14.0-arm64-cpython-38/Bio
     879      creating build/temp.macosx-14.0-arm64-cpython-38/Bio/Align
     880      /Users/luke/Documents/GitHub/luke-dt/spack/lib/spack/env/clang/clang -Wno-unused-result -Wsign-compare -Wunreachable-code 
            -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/Users/luke/Documents/GitHub/luke-dt/spack/opt/spack/darwin-sonoma-m1/apple-clang-15.0
            .0/python-venv-1.0-f5gpbirgzftck6jidbsau2d3cx5osrvw/include -I/Users/luke/Documents/GitHub/luke-dt/spack/opt/spack/darwin-so
            noma-m1/apple-clang-15.0.0/python-3.8.19-rvnqtxp32jvar6dk66tyrjlu6xhylalc/include/python3.8 -c Bio/Align/_aligners.c -o buil
            d/temp.macosx-14.0-arm64-cpython-38/Bio/Align/_aligners.o
     881      In file included from Bio/Align/_aligners.c:11:
  >> 882    /Users/luke/Documents/GitHub/luke-dt/spack/opt/spack/darwin-sonoma-m1/apple-clang-15.0.0/python-3.8.19-rvnqtxp32jvar6dk66tyr
            jlu6xhylalc/include/python3.8/Python.h:44:10: fatal error: 'crypt.h' file not found
     883      #include <crypt.h>
     884               ^~~~~~~~~
     885      1 error generated.
     886      error: command '/Users/luke/Documents/GitHub/luke-dt/spack/lib/spack/env/clang/clang' failed with exit code 1
     887      error: subprocess-exited-with-error
     888    
</pre></details>


I couldn't find info on spack, but from googling it appears that crypt.h is no longer included in MacOS:

https://groups.google.com/a/soe.ucsc.edu/g/genome-mirror/c/xc0A1dM3ndg

I understand if this is a bug that should be fixed elsewhere, but this is what allowed me to build it on my machine. Here is the spec for reference:

```console
luke@lukes-mbp ~ % spack spec py-bakta
Input spec
--------------------------------
 -   py-bakta

Concretized
--------------------------------
 -   py-bakta@1.5.1%apple-clang@15.0.0~deepsig build_system=python_pip arch=darwin-sonoma-m1
[+]      ^amrfinder@3.12.8%apple-clang@15.0.0 build_system=makefile arch=darwin-sonoma-m1
[+]          ^curl@8.6.0%apple-clang@15.0.0~gssapi~ldap+libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=secure_transport arch=darwin-sonoma-m1
[+]              ^libidn2@2.3.4%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]                  ^libunistring@1.1%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]              ^nghttp2@1.57.0%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]          ^gmake@4.4.1%apple-clang@15.0.0~guile build_system=generic arch=darwin-sonoma-m1
[+]      ^aragorn@1.2.41%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]      ^blast-plus@2.16.0%apple-clang@15.0.0+bzip2+freetype+gnutls+jpeg+lzo+openmp+openssl+pcre+perl+png+python~static+zlib build_system=autotools arch=darwin-sonoma-m1
[+]          ^bzip2@1.0.8%apple-clang@15.0.0~debug~pic+shared build_system=generic arch=darwin-sonoma-m1
[+]              ^diffutils@3.10%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]          ^cpio@2.15%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]          ^freetype@2.13.2%apple-clang@15.0.0+pic+shared build_system=autotools arch=darwin-sonoma-m1
[+]          ^gnuconfig@2022-09-17%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]          ^gnutls@3.8.3%apple-clang@15.0.0~guile+zlib build_system=autotools arch=darwin-sonoma-m1
[+]              ^nettle@3.9.1%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]                  ^gmp@6.2.1%apple-clang@15.0.0+cxx build_system=autotools libs=shared,static patches=69ad2e2 arch=darwin-sonoma-m1
[+]          ^libjpeg-turbo@3.0.3%apple-clang@15.0.0~ipo~jpeg8~partial_decoder+pic build_system=cmake build_type=Release generator=make libs=shared,static arch=darwin-sonoma-m1
[+]              ^nasm@2.15.05%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]          ^libpng@1.6.39%apple-clang@15.0.0~ipo~pic build_system=cmake build_type=Release generator=make libs=shared,static arch=darwin-sonoma-m1
[+]          ^llvm-openmp@18.1.0%apple-clang@15.0.0~ipo+multicompat build_system=cmake build_type=Release generator=make arch=darwin-sonoma-m1
[+]          ^lmdb@0.9.31%apple-clang@15.0.0 build_system=makefile arch=darwin-sonoma-m1
[+]          ^lzo@2.10%apple-clang@15.0.0 build_system=autotools libs=shared,static arch=darwin-sonoma-m1
[+]          ^openssl@3.2.1%apple-clang@15.0.0~docs+shared build_system=generic certs=mozilla arch=darwin-sonoma-m1
[+]              ^ca-certificates-mozilla@2023-05-30%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]          ^pcre@8.45%apple-clang@15.0.0~jit+multibyte+pic+shared+static+utf build_system=autotools arch=darwin-sonoma-m1
[+]          ^perl@5.38.0%apple-clang@15.0.0+cpanm+opcode+open+shared+threads build_system=generic arch=darwin-sonoma-m1
[+]              ^berkeley-db@18.1.40%apple-clang@15.0.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=darwin-sonoma-m1
[+]          ^zlib@1.3%apple-clang@15.0.0+optimize+pic+shared build_system=makefile arch=darwin-sonoma-m1
[+]      ^diamond@2.1.9%apple-clang@15.0.0+blast~eigen~ipo+zstd build_system=cmake build_type=Release generator=make arch=darwin-sonoma-m1
[+]          ^cmake@3.30.2%apple-clang@15.0.0~doc+ncurses+ownlibs build_system=generic build_type=Release patches=dbc3892 arch=darwin-sonoma-m1
[+]          ^zstd@1.5.5%apple-clang@15.0.0+programs build_system=makefile compression=none libs=shared,static arch=darwin-sonoma-m1
[+]      ^hmmer@3.4%apple-clang@15.0.0~gsl+mpi build_system=generic arch=darwin-sonoma-m1
[+]          ^openmpi@5.0.5%apple-clang@15.0.0~atomics~cuda~gpfs~internal-hwloc~internal-libevent~internal-pmix~java~legacylaunchers~lustre~memchecker~openshmem~orterunprefix~romio+rsh~static+vt+wrapper-rpath build_system=autotools fabrics=none romio-filesystem=none schedulers=none arch=darwin-sonoma-m1
[+]              ^automake@1.16.5%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]              ^hwloc@2.9.3%apple-clang@15.0.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl~pci~rocm build_system=autotools libs=shared,static arch=darwin-sonoma-m1
[+]              ^libevent@2.1.12%apple-clang@15.0.0+openssl build_system=autotools arch=darwin-sonoma-m1
[+]              ^libtool@2.4.7%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]                  ^findutils@4.9.0%apple-clang@15.0.0 build_system=autotools patches=440b954 arch=darwin-sonoma-m1
[+]              ^openssh@9.7p1%apple-clang@15.0.0+gssapi build_system=autotools patches=3505c58,d886b98 arch=darwin-sonoma-m1
[+]                  ^krb5@1.20.1%apple-clang@15.0.0+shared build_system=autotools arch=darwin-sonoma-m1
[+]                      ^bison@3.8.2%apple-clang@15.0.0~color build_system=autotools arch=darwin-sonoma-m1
[+]                  ^libedit@3.1-20230828%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]              ^pmix@5.0.2%apple-clang@15.0.0~docs+pmi_backwards_compatibility~python~restful build_system=autotools arch=darwin-sonoma-m1
[+]      ^infernal@1.1.5%apple-clang@15.0.0~mpi build_system=autotools arch=darwin-sonoma-m1
[+]      ^pilercr@1.06%apple-clang@15.0.0 build_system=makefile arch=darwin-sonoma-m1
[+]      ^prodigal@2.6.3%apple-clang@15.0.0 build_system=makefile arch=darwin-sonoma-m1
[+]      ^py-alive-progress@1.6.2%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
 -       ^py-biopython@1.81%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]          ^py-numpy@1.24.4%apple-clang@15.0.0 build_system=python_pip patches=873745d arch=darwin-sonoma-m1
[+]              ^openblas@0.3.26%apple-clang@15.0.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=none arch=darwin-sonoma-m1
[+]              ^py-cython@0.29.36%apple-clang@15.0.0 build_system=python_pip patches=c4369ad arch=darwin-sonoma-m1
[+]      ^py-pip@23.1.2%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]      ^py-requests@2.32.2%apple-clang@15.0.0~socks build_system=python_pip arch=darwin-sonoma-m1
[+]          ^py-certifi@2023.7.22%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]          ^py-charset-normalizer@3.3.0%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]          ^py-idna@3.4%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]              ^py-flit-core@3.9.0%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]          ^py-urllib3@2.1.0%apple-clang@15.0.0~brotli~socks build_system=python_pip arch=darwin-sonoma-m1
[+]              ^py-hatchling@1.21.0%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]                  ^py-editables@0.5%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]                  ^py-pathspec@0.11.1%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]                  ^py-pluggy@1.5.0%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]                  ^py-trove-classifiers@2023.8.7%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]                      ^py-calver@2022.6.26%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]      ^py-setuptools@63.4.3%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]      ^py-wheel@0.41.2%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]      ^py-xopen@1.6.0%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]          ^py-setuptools-scm@8.0.4%apple-clang@15.0.0+toml build_system=python_pip arch=darwin-sonoma-m1
[+]              ^git@2.45.2%apple-clang@15.0.0+man+nls+perl+subtree~svn~tcltk build_system=autotools arch=darwin-sonoma-m1
[+]                  ^pcre2@10.42%apple-clang@15.0.0~jit+multibyte build_system=autotools arch=darwin-sonoma-m1
[+]              ^py-packaging@23.1%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]              ^py-tomli@2.0.1%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]              ^py-typing-extensions@4.8.0%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]      ^python@3.8.19%apple-clang@15.0.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,4c24573,ebdca64,f2fd060 arch=darwin-sonoma-m1
[e]          ^apple-libuuid@1353.100.2%apple-clang@15.0.0 build_system=bundle arch=darwin-sonoma-m1
[+]          ^expat@2.6.2%apple-clang@15.0.0~libbsd build_system=autotools arch=darwin-sonoma-m1
[+]          ^gdbm@1.23%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]          ^gettext@0.22.4%apple-clang@15.0.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=darwin-sonoma-m1
[+]              ^libiconv@1.17%apple-clang@15.0.0 build_system=autotools libs=shared,static arch=darwin-sonoma-m1
[+]              ^libxml2@2.10.3%apple-clang@15.0.0+pic~python+shared build_system=autotools arch=darwin-sonoma-m1
[+]              ^tar@1.34%apple-clang@15.0.0 build_system=autotools zip=pigz arch=darwin-sonoma-m1
[+]                  ^pigz@2.8%apple-clang@15.0.0 build_system=makefile arch=darwin-sonoma-m1
[+]          ^libffi@3.4.4%apple-clang@15.0.0 build_system=autotools patches=070b1f3 arch=darwin-sonoma-m1
[+]          ^libxcrypt@4.4.35%apple-clang@15.0.0~obsolete_api build_system=autotools patches=4885da3 arch=darwin-sonoma-m1
[+]          ^ncurses@6.4%apple-clang@15.0.0~symlinks+termlib abi=none build_system=autotools arch=darwin-sonoma-m1
[+]          ^pkgconf@1.9.5%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]          ^readline@8.2%apple-clang@15.0.0 build_system=autotools patches=bbf97f1 arch=darwin-sonoma-m1
[+]          ^sqlite@3.43.2%apple-clang@15.0.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=darwin-sonoma-m1
[+]          ^xz@5.4.6%apple-clang@15.0.0~pic build_system=autotools libs=shared,static arch=darwin-sonoma-m1
[+]      ^python-venv@1.0%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]      ^trnascan-se@2.0.12%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]          ^autoconf@2.72%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]              ^m4@1.4.19%apple-clang@15.0.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=darwin-sonoma-m1
[+]                  ^libsigsegv@2.14%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
```

File: var/spack/repos/builtin/packages/blast-plus/package.py (hunk 2 above)
Comment (luke-dt) [reply var/spack/repos/builtin/packages/blast-plus/package.py]: yeah, just an oversight on my part. the package configure file adds openmp by default, but it wouldn't need this dependency if you ask to build without openmp obviously.

File: var/spack/repos/builtin/packages/py-biopython/package.py (hunk 3 above)
Comment (RMeli) [reply var/spack/repos/builtin/packages/py-biopython/package.py]: I'm not sure what is going on here, because I can't reproduce it locally. I never had issues installing `biopython` on macOS 14 with `%apple-clang` and I just tried on macOS 15 and it worked well: `py-biopython %apple-clang@16`.

The issue you linked is from 2012, so I'm not sure what to make of it.

Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/py-biopython/package.py]: @luke-dt About the error that you report here:
```py
     881 In file included from Bio/Align/_aligners.c:11:
  >> 882 .../spack/darwin-sonoma-m1/apple-clang-15.0.0/python-3.8.19-.../include/python3.8/Python.h:44:10:
   fatal error: 'crypt.h' file not found
```
So `py-biopython` includes a `Python.h` header from Python 3.8 that has been installed by `apple-clang-15.0.0`.

I don't know about MacOS specifics, but Python 3.8 has officially reached [EOL status](https://devguide.python.org/versions/). Since it is no longer supported by PSF and no longer receives fixes to security vulnerabilities, it should be deprecated.

Adam already opened a PR to start the move from removing support from it in `spack` away from Python 3.8 by deprecating: https://github.com/spack/spack/pull/46913

It will still be some work to get all packages away from 3.8, but the work is ongoing.

That means when this PR successfully merged (maybe after the upcoming `spack` release or maybe before), you'd have to use `--deprecated` when installing packages that need it.

You might therefore consider re-installing the spec dependency `^apple-clang` using a newer Python version, so at least Python 3.9 by appending `^apple-clang^python@3.9` to the spec you build anyway at some point.

And, looking of a newer version of `apple-clang` may also be worth investigating. Again, I have no idea on macOS or bio specifics, but `spack` develop nowadays often decides to build llvm@17 and it already has llvm+clang@19 in the llvm package on Linux.

Just as a data point: LLVM-15 was the first LLVM release that can be built without distutils that was removed from Python with Python 3.12, so it is the first release after that was updated to support the Python version(s) that are currently the latest versions.

Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/py-biopython/package.py]: @luke-dt BTW, what about #46978:

That is the same PR (maybe without some minor changes) but it does not involve biopython.

Usually, it is easier to address and merge changes that only involve one package like you propose in #46978 and address things incrementally instead.

For #46978, you only need the approval of @weijianwen (or remind us again about it if he did not respond in a week) to get the first part of your changes merged, and by then, maybe the issue could be solved by moving to newer versions of Python and apple-clang, and you are not producing this on a version that has reached EOL and is about to be deprecated in `spack` too soonish.

Comment (luke-dt) [reply var/spack/repos/builtin/packages/py-biopython/package.py]: I suspected that python 3.8 might have something to do with it, since all I found by searching the spack issues was #41014. Unfortunately, when I try to build the package [I'm actually trying to update](https://github.com/luke-dt/spack/tree/package/py-bakta-194), spack refuses to concretize:

<pre>
luke@lukes-mbp ~ % spack spec py-bakta^python@3.10
==> Error: failed to concretize `py-bakta ^python@3.10` for the following reasons:
     1. Cannot select a single "version" for package "py-bakta"
luke@lukes-mbp ~ % spack spec py-bakta^python@3.9 
==> Error: failed to concretize `py-bakta ^python@3.9` for the following reasons:
     1. Cannot select a single "version" for package "py-bakta"
</pre>

I can update the PR to be similar to what you requested, that's fine. But if you have any insight into why spack is behaving this way, I would appreciate it.

Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/py-biopython/package.py]: @luke-dt On the PR: @RMeli's answer (and the log that it's LLVM's header that causes it, not biopython) that includes the crypt header, fixing/diagnosing this would basically the responsibility of the apple-clang setup, not of a user package like biopython that should not have anything to do with what includes LLVM has.

Because of that so far not yet understood issue, I'd remove this part from this PR:
```py
 depends_on("libxcrypt", when="%apple-clang")
```
If you can add `^libxcrypt` to your spec and that works as a workaround for now, I'd tend to do that until the correct fix has been found.

- The next question to ask would be if you can use a newer apple-clang and/or if you can re-build the Python build that is used (just in case).

- The discussion on Python 3.9 is more longterm as Python 3.8 will likely (at least possibly) be deprecated in `spack` at some point. To be ready when that occurs, it likey best to check if `bakta` can actually work with Python 3.9. I'm not aware of the changes between 3.8 and 3.9, so I would hope that it should work at runtime if the installation can be updated to succeed.

You apparently would need to bring two components together to try building by-backta with `^python3.9`.
- Some of the changes from #46913 (I just switched to this branch, could not see which parts you'd need)
- Update `py-deepsig-biocomp`, it has multiple dependency versions pinned to a single version which are likely causes by too strict judgement by the upstream project. It's not active anymore and even on pypi.org, the Pthon3.8 version dependency is conflicting. At one place it is just 3.8 only, but in the metadata next to the package itself it clearly says "Python >= 3.8". I assume the same incorrect strictness for the other pinned versions:

```py
--- a/var/spack/repos/builtin/packages/py-deepsig-biocomp/package.py
+++ b/var/spack/repos/builtin/packages/py-deepsig-biocomp/package.py
@@ -19,11 +19,11 @@ class PyDeepsigBiocomp(PythonPackage):

     version("1.2.5", sha256="e954b815d63c221c564c7d3fe27123d7cd2c39b191d6107369ab095d506496e0")

-    depends_on("python@3.8", type=("build", "run"))
+    depends_on("python@3.8:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
     depends_on("py-biopython@1.78:", type=("build", "run"))
-    depends_on("py-keras@2.4.3", type=("build", "run"))
-    depends_on("py-tensorflow@2.2.0", type=("build", "run"))
+    depends_on("py-keras@2.4.3:", type=("build", "run"))
+    depends_on("py-tensorflow@2.2.0:", type=("build", "run"))
     depends_on("py-tensorboard", type=("build", "run"))
```
On a branch of #46913 with this fix applied, apparently (unless I forgot something) I could concretize `py-bakta^python@3.9.`

Comment (RMeli) [inline var/spack/repos/builtin/packages/py-biopython/package.py]:
```suggestion
```

>  fixing/diagnosing this would basically the responsibility of the apple-clang setup, not of a user package like biopython that should not have anything to do with what includes LLVM has.

I agree with this assessment. I did not see anything in the `biopython` repository indicating that this is a dependency of the package.

**Review by RMeli — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-biopython/package.py (hunk 3 above)
Comment (luke-dt) [reply var/spack/repos/builtin/packages/py-biopython/package.py]: Thanks for digging in to this for me! I am happy making the changes you suggested since it appears that `crypt.h` is an issue with the apple-clang setup, not the biopython package.

I also appreciate your suggestions to the `py-deepsig-biocomp` package, and I agree with your assessment (overly-strict requirements causing conflicts). The only thing that I think is different between our machines is that I'm using `darwin` so when concretizing `py-bakta^python@3.9` on my machine, `py-bakta` is building without `py-deepsig-biocomp` anyway and insists that it needs python 3.8. I would love to move away from this older version of python, so is there a way to deduce from spack what dependencies are restricting the python version to 3.8?

Otherwise, I'll try using an updated version of `apple-clang` and see if that resolves things.

Comment (luke-dt) [reply var/spack/repos/builtin/packages/py-biopython/package.py]: Also sorry about the duplicate PR (#46978), I absentmindedly duplicated the `blast-plus` changes on this branch also. I'll remove it here.

**Review by RMeli — APPROVED** [review]
Checksum verified. LGTM.

**Review by tldahlgren — APPROVED** [review]
